.github/workflows/release-core.yml

```diff
@@ -10,88 +10,124 @@ on:
+        # cibuildwheel's repair step in delocation. This seems to be because
+        # the Swift standard libraries aren't stored in /usr/lib/swift/ for
+        # x86_64 architecture.
+        #
+        # You can test this by compiling AppleStats for arm64 and x86_64
+        # separately, and then running "otool -L AppleStats". On x86_64,
+        # you'll see entries like "@rpath/libswiftCore.dylib", whereas on
+        # arm64 they are "/usr/lib/swift/libswiftCore.dylib".
+        #
+        # This could just be a Swift compiler bug.
+        #
+        # Interestingly, "/usr/lib/swift/libswiftCore.dylib" doesn't exist
+        # under /usr/lib/swift/ on my machine, yet the binary works.
+        #
+        # The "delocate" step copies non-system dependencies into the wheel.
+        # It explicitly ignores "/usr/lib/..." paths, but when it sees
+        # "@rpath/libswiftCore.dylib" it tries to find the binary. The RPATH
+        # on x86_64 includes "/usr/lib/swift/", but like I mentioned, the
+        # files don't actually exist, and the step fails.
+      - name: Build artifacts using wini (macOS-13)
+        if: matrix.os == 'macOS-13'
+        run: bash wini build wandb-core-artifacts --skip-swift-build
 
+      ##################################################
+      # All platforms: create wheels
+      ##################################################
       - name: Build wheels
         uses: pypa/cibuildwheel@v2.16.5
         with:
-           package-dir: "core"
-           output-dir: "core/wheelhouse"
+           package-dir: core
+           output-dir: core/wheelhouse
         env:
-          CIBW_ENVIRONMENT: PATH=$PATH:/usr/local/go/bin
-          CIBW_BEFORE_ALL: >
-            pip install -r requirements_tools.txt &&
-            bash wini build wandb-core-artifacts --skip-swift-build
+          # In cp36-*, the wheel name sometimes includes additional dashes that
+          # make it invalid, breaking the job.
+          #
+          # Not sure why we skip PyPy and musllinux builds.
+          #
+          # See https://cibuildwheel.readthedocs.io/en/stable/options/#build-skip
+          CIBW_SKIP: cp36-* pp* *musllinux*
+          CIBW_ARCHS_LINUX: x86_64 aarch64
+          # Linux is special because it's built in a container using qemu, so we must
+          # build our Go binary inside the cibuildwheel step.
+          #
+          # Some downsides:
+          #   - We have to download Go manually
+          #   - We hard-code the Go version instead of reading it from go.mod
+          #   - We don't get caching
           CIBW_BEFORE_ALL_LINUX: >
-            python core/scripts/build/install_go.py &&
+            case $(uname -m) in
+              aarch64) export DOWNLOAD_GOARCH=arm64;;
+              x86_64) export DOWNLOAD_GOARCH=amd64;;
+            esac &&
+            curl -L https://golang.org/dl/go1.22.0.linux-$DOWNLOAD_GOARCH.tar.gz > go.tar.gz &&
```

Comment:
can we make the go version a parameter so it is consistent with our  mod file's version

Reply:
updated to use `grep '^go ' core/go.mod | cut -d' ' -f2` which looks "readable" to me, underscoring the negative impact of bash-in-YAML devops on the human mind